### PR TITLE
[DOCS][REFACTOR] Reorganize the docs.

### PR DESCRIPTION
**Pull request by tqchen**
- Move most toctree to `:hiden:` so there can be top-level categorizations in the navigation bar.
- Move frontend guide into design and developer guides
- Move get started tutorials into its separate folder.

Co-authored-by: Chris Hoge <chris@hogepodge.com>

### CODE REVIEW

**Comment by tqchen** [comment]
cc @hogepodge @jroesch @zhiics @yzhliu @tmoreau89

File: docs/index.rst [hunk 1]
```diff
@@ -18,43 +18,75 @@
 TVM Documentation
 =================
 
+TVM is an open deep learning compiler stack for CPUs, GPUs, and specialized accelerators. It aims to close the gap between the productivity-focused deep learning frameworks, and the performance- or efficiency-oriented hardware backends.
```
Comment (binarybana) [inline docs/index.rst:21]: Can we expand this section a bit? (or are you open for commits doing so?). Eg: what kind of users benefit from TVM, what can I do with it, what benefits does it have over the competition (at a high level).

**Comment by binarybana** [comment]
Left one comment, otherwise looking great!

File: docs/index.rst (hunk 1 above)
Comment (tqchen) [reply docs/index.rst:21]: We can discuss whether the content belongs to the landing page (in the main website) or here. One thing that I think we could further improve on are possible pointers(e.g. what are the things that i need to read wearing hat of different personas)

**Pull request merged**